### PR TITLE
Should fix https://github.com/rackspace-cookbooks/phpstack/issues/285

### PR DESCRIPTION
**Pull request by jujugrrr**
Required to get the latest version of `application`.
I've removed support for application_php as it was preventing to upgrade `application`, and it not maintained anymore.

I don't think the attribute package was used by any customer cookbook : 
[search packages](https://github.com/search?utf8=%E2%9C%93&q=user%3AAutomationSupport+%22%5B%27deployment%27%5D%5B%27packages%27%5D%22&type=Code&ref=searchresults)
contrary to before_symlink by example : 
[search before_symlink](
https://github.com/search?q=user%3AAutomationSupport+%22%5B%27deployment%27%5D%5B%27before_symlink_script_name%27%5D%22&type=Code&utf8=%E2%9C%93).

@mattjbarlow You highlighted the possible issue with the `packages` parameter, maybe you could review this PR?

@marcoamorales I think you are the first one to have got the issue, could you give it a try?

Thank you!

### CODE REVIEW

File: recipes/application_php.rb [hunk 1]
```diff
@@ -115,7 +115,7 @@
           before_symlink site_opts['deployment']['before_symlink_script_name']
         end
         # add in all of the other application resource attributes that aren't being defined
-        %w(packages keep_releases strategy scm_provider rollback_on_error environment purge_before_symlink
+        %w(keep_releases strategy scm_provider rollback_on_error environment purge_before_symlink
```
Comment (prometheanfire) [inline recipes/application_php.rb]: removing the packages function?

File: metadata.rb [hunk 2]
```diff
@@ -8,8 +8,7 @@
 version '3.0.25'
 
 depends 'apache2'
-depends 'application'
-depends 'application_php'
+depends 'application', '~> 4.1.6'
```
Comment (prometheanfire) [inline metadata.rb]: if removing application_php allows us to upgrade, why do we need to pin?  also, why are we pinning to such a specific version?

**Comment by jujugrrr** [comment]
I was not sure about this one indeed (see description), it seems to come from application_php, I don't see any customer using this feature (and there is `php_pear` resource doing the same thing), and it's the only thing relating to application_php?

ps : have look at my description, I'm happy to get input from you about it

**Comment by jujugrrr** [comment]
Ah my bad packages is in the default application as well. I'm chaing it back.

By the way, when have we stopped to run test-kitchen on Jenkins?

**Comment by prometheanfire** [comment]
just because no one is using it doesn't mean we remove it though

**Comment by jujugrrr** [comment]
> just because no one is using it doesn't mean we remove it though
I thought it was for php_pear package only, I misread it.

Do you want me still to keep application_php?

**Comment by prometheanfire** [comment]
ah, it wasn't obvious from the comments that it was provided by that cookbook, ok, remove it then.  What about the pinning though, why are we doing that?

**Comment by jujugrrr** [comment]
> why are we doing that?

I pinned it because whatever I tried Berkshelf kept wanting install application 3.0.0, @martinb3 gave it a try without success as well.

**Comment by jujugrrr** [comment]
@prometheanfire if you can give it a try on your side it would be nice, as martin and I might have missed something.
I'll catch up monday

**Comment by prometheanfire** [comment]
can we change it to >= instead then?

File: recipes/application_php.rb (hunk 1 above)
Comment (martinb3) [inline recipes/application_php.rb]: This one seems like a bug; do we really want to remove this?

File: metadata.rb (hunk 2 above)
Comment (martinb3) [inline metadata.rb]: (a) Berkshelf is depsolving by downgrading application, so that's the need to pin. (b) I'd be okay with a `>=` pin.

**Comment by martinb3** [comment]
@jujugrrr once you fix the packages thing, LGTM to merge. I'd recommend the change to `>=` as well.

**Comment by prometheanfire** [comment]
packages is a part of the regular app cookbook, so no need to remove it.

>= please :D

**Comment by jujugrrr** [comment]
thanks @martinb3

**Pull request merged**